### ISSUE TITLE
[patch] Detect kernels from opencl.kernels MD

### ISSUE THREAD

**Issue by pjaaskel** [HSAFoundation/HLC-HSAIL-Development-LLVM#18]
This patch adds detection of kernels from the opencl.kernels metadata: https://github.com/pocl/pocl/blob/master/tools/patches/llvm-3.7-hsail-branch.patch

**Comment by arsenm** [comment]
Do you think it would be better if we had an OpenCLMetadataAnalysis pass that we could query for this and all of the other OpenCL metadata?

**Comment by pjaaskel** [comment]
Yes that would definitely be cleaner as there's plenty of other OpenCL/SPIR metadata.
